common/logger/src/struct_log.rs

```diff
@@ -426,108 +426,122 @@ impl FileStructLogThread {
 
 /// Sink that streams all structured logs to an address through TCP protocol
 struct TCPStructLog {
-    sender: SyncSender<StructuredLogEntry>,
+    sender: SyncSender<String>,
 }
 
 impl TCPStructLog {
-    pub fn start_new(address: String) -> io::Result<Self> {
+    pub fn start_new(endpoint: String) -> io::Result<Self> {
         let (sender, receiver) = mpsc::sync_channel(WRITE_CHANNEL_SIZE);
-        let sink_thread = TCPStructLogThread::new(receiver, address);
+        let sink_thread = TCPStructLogThread::new(receiver, endpoint);
         thread::spawn(move || sink_thread.run());
         Ok(Self { sender })
     }
 }
```

Comment:
This may be a naive question but is there a particular reason why we can't write directly to stdout/stderr and instead just have stdout/stderr itself somehow redirected to where ever it needs to go? In other words make this problem, not our software's issue but some other piece of infra's problem? Maybe this sort of goes back to writing to a file instead...

Reply:
That does go back to the writing to a file instead.  The TCP log writing is only written specifically to ship the logs to a different container, which is an infrastructure problem.

We could write to stdout, and somehow take kubernetes logs and spew them into elasticsearch with either Logstash or Vector.